broadinstitute/ddp-study-server
# DDP-8651 [DSM] running DSM tests in circleCi

This enables the DSM test suite to be run as part of the combined test workflow in circleCI.  It currently excludes the tests that require specific environment setup, or otherwise failed when I ran them in the CI environment.  Future work can handle upgrading those tests so they pass in CI.   The Maven pom variables are set up so that running `mvn test` locally includes all tests, and you can run `mvn test -Ddsm.test.excludes='${dsm.test.ci.excludes}'` to locally run only the tests that would run in CI.

TO TEST:
1. run `mvn test`, confirm tests run
2. run `mvn test -Ddsm.test.excludes='${dsm.test.ci.excludes}'`  confirm fewer tests run (and all should pass)
3. in the 'checks' for this PR , confirm that the required checks all passed 
   Inspect    'combined-build-test-workflow' 
                   -> dsm-compile-and-run-tests-job
                    -> Run Maven tests with options
      you should see a console log indicating tests were run:
                    `Tests run: 226, Failures: 0, Errors: 0, Skipped: 5`
